# Primeira Questão

A partir de uma tabela que mostra a distribuição de frequência de mulheres, na faixa etária
de 25 a 34 anos, em relação a pressão sistólica em milímetros de mercúrio, será respondido algumas questões.

# Bibliotecas

Primeiramente, é necessário realizar a importação das dependências que serão utilizadas no código.


In [12]:
import pandas as pd
import matplotlib.pyplot as plt

In [22]:
df = pd.read_csv('q1.csv',delimiter=',')
df

pressao_siatolica(mm de Hg)  frequencia  ponto_medio
0                 [80 |--100)          30           90
1                 (100|--120)          80          110
2                 (120|--140)          40          130
3                 (140|--160)          10          150
4                 (160|--180)           4          170
5                 (180|--200]           6          190

Foi realizada a adição do ponto médio a tabela de distribuição. O  **ponto_medio** é a soma entre os intervalos dividido por 2.

Abaixo se encontra o cálculo da porcentagem de cada frequência a tabela. Essa porcentagem da frequência será adicionada a tabela.

In [23]:
df['porcentagem'] = (df['frequencia'] /df['frequencia'].sum() )*100
df.round(3)

pressao_siatolica(mm de Hg)  frequencia  ponto_medio  porcentagem
0                 [80 |--100)          30           90       17.647
1                 (100|--120)          80          110       47.059
2                 (120|--140)          40          130       23.529
3                 (140|--160)          10          150        5.882
4                 (160|--180)           4          170        2.353
5                 (180|--200]           6          190        3.529

**a)** 
A percentagem de mulheres que têm pressão igual ou maior que 120mm de Hg é de  **17,64%**


**b)** O número de mulheres que têm pressão menor que 160 mm de Hg é de **155 mulheres.** Para chegar a essa resposta foram somandos as frequência das linhas _0_ a _3_, sendo a última só sendo considerado metade do valor, ou seja 5. 

**c)** O número de mulheres (e a percentagem) que têm pressão igual ou maior que
100mm de Hg é de **140** com a porcentagem de **82,32%**. Esses resultados advém da soma da frequência e da porcentagem , respectivamente, das linhas _1_ até _5_.

**d)** A pressão sistólica mais frequente, a partir do cálculo da moda de um agrupamento de dados, é:  



\begin{equation*}
Mo = Lmd+ \frac {D1}{D1+D2} * h
\end{equation*}



Primeiramente, a classe modal é 80(linha 2). Em relação a fórmula **Lmd** é o limite inferior da classe modal. Já **D1** é a _frequência absoluta menos a frequência anterior da classe modal_. Enquanto **D2** é a _frequência absoluta menos a frequência posterior da classe modal_. Por fim, **h** é a amplitude da classe modal.

O valor de **Lmd** é 100. O valor de **D1=50**(80-30); **D2=40**(80-40). Sendo assim, o valor de **D1+D2=90**. Já o valor de **h=20**. A fórmula com a substituição dos valores fica da seguinte forma: 


\begin{equation*}
Mo = 100+ \frac {50}{90} * 20
\end{equation*}

Em que o resultado é **Mo = 111,11** 

**d)** O desvio padrão e o coeficiente de variação para a pressão sistólica.

Antes de realizar esses calculos é preciso adicionar alguns valores a tabela de distribuição. **frenquência** receberá mometaneamente a denominação de **fi**, enquanto o **ponto_medio** de **xi**. Será adicionado um coluna(**xi.fi**) que resulta da multiplicação da **frequencia** com o **ponto_medio**.

In [24]:
df['xi.fi'] = df['frequencia']*df['ponto_medio']
df.round(3)

pressao_siatolica(mm de Hg)  frequencia  ponto_medio  porcentagem  xi.fi
0                 [80 |--100)          30           90       17.647   2700
1                 (100|--120)          80          110       47.059   8800
2                 (120|--140)          40          130       23.529   5200
3                 (140|--160)          10          150        5.882   1500
4                 (160|--180)           4          170        2.353    680
5                 (180|--200]           6          190        3.529   1140

**xi²** é o resultado do **ponto_medio** ao quadrado. 

In [25]:
df['xi²'] = (df['ponto_medio']*df['ponto_medio'])
df.round(3)

pressao_siatolica(mm de Hg)  frequencia  ponto_medio  porcentagem  xi.fi  \
0                 [80 |--100)          30           90       17.647   2700   
1                 (100|--120)          80          110       47.059   8800   
2                 (120|--140)          40          130       23.529   5200   
3                 (140|--160)          10          150        5.882   1500   
4                 (160|--180)           4          170        2.353    680   
5                 (180|--200]           6          190        3.529   1140   

     xi²  
0   8100  
1  12100  
2  16900  
3  22500  
4  28900  
5  36100

**xi².fi** é a multiplicação da frequência pelo quadrado do ponto médio.

In [26]:
df['xi².fi'] = (df['xi²']*df['frequencia'])
df

pressao_siatolica(mm de Hg)  frequencia  ponto_medio  porcentagem  xi.fi  \
0                 [80 |--100)          30           90    17.647059   2700   
1                 (100|--120)          80          110    47.058824   8800   
2                 (120|--140)          40          130    23.529412   5200   
3                 (140|--160)          10          150     5.882353   1500   
4                 (160|--180)           4          170     2.352941    680   
5                 (180|--200]           6          190     3.529412   1140   

     xi²  xi².fi  
0   8100  243000  
1  12100  968000  
2  16900  676000  
3  22500  225000  
4  28900  115600  
5  36100  216600

A fórmula utilizada para calcular o desvio padrão é a seguinte: 

\begin{equation*}
d = \sqrt{ \frac{1}{N}*[ \sum (xi².fi) - \frac{(\sum xi.fi)²}{N} ]}
\end{equation*}


Substituindo os valores na equação, em que N é o valor total da **frequencia**. Temos: 

\begin{equation*}
d = \sqrt{ \frac{1}{170}*[ 2444200 - (\frac{(20020)²}{170}) ]}
\end{equation*}


O resultado dessa equação é **22,56**. Ou seja, desvio padrão = **22,56**.

Para calcular a **coeficiente de variação** é necessário ter a média, que neste caso, é um cálculo entre a **frequencia e o ponto_medio**. 

\begin{equation*}
\bar{x}= \frac {(90*30)  +  (110*80) + (130*40) + (150*10) + (170*4) + (190*6) }{170} 
\end{equation*}

E o resultadado da **média = 117,76**


Então o coeficiente de varição é :

\begin{equation*}
cv = \frac {d}{s} 
\end{equation*}

Substituindo os valores de **d** pelo desvio padrão e de **x-barra** pelo valor da média , temos:

\begin{equation*}
cv = \frac {22,56}{117,76} 
\end{equation*}


O valor de **cv = 0,19157** ou **19,16%**